addons/purchase/models/purchase_order_line.py

```diff
@@ -86,6 +87,20 @@ class PurchaseOrderLine(models.Model):
             "Forbidden values on non-accountable purchase order line"),
     ]
 
+    @api.depends_context('partner_id')
+    def _compute_partner_reference(self):
+        for line in self:
+            product = line.product_id
+            product_name = ""
+            product_code = ""
+            for supplier in product.seller_ids:
+                if supplier.partner_id == line.order_id.partner_id:
+                    if supplier.product_name:
+                        product_name = supplier.product_name
+                    if supplier.product_code:
+                        product_code = supplier.product_code
+            line.partner_reference = '%s%s' % (product_code and '[%s] ' % product_code or '', product_name)
+
```

Comment:
I don't see this needed in the spec.

Reply:
@Whenrow 
![task_added](https://github.com/odoo/odoo/assets/157007156/0ce477fb-7e74-428b-a196-c4e266f87c6a)
While i was doing this task thd asked me to cover this task:
https://www.odoo.com/web#id=3673331&cids=1&menu_id=4722&action=333&active_id=809&model=project.task&view_type=form
so the specs related to this code is mentioned in the above shared link.
Thanks

Reply:
@vagu-odoo I would make sure the product code and product description is always filled on the purchase order line when created automatically

Reply:
@vagu-odoo Do we still need the compute method if the code and description are always set in description ?

Reply:
Hello @Whenrow 

Yaa, it looks like there is no need for a compute method because the vendor product code and name are already included in the description field. So, I have removed the compute and let skip that point. Let me know if I misunderstood anything.

Thanks.